### PR TITLE
Update to pac4j 6.0.0

### PR DESCRIPTION
**Pull request by rnett**
For internal reviews

Demo/test companion: https://github.com/rnett/ratpack-pac4j/pull/2

### CODE REVIEW

File: src/main/java/ratpack/pac4j/RatpackPac4j.java [hunk 1]
```diff
@@ -350,24 +360,25 @@ public static Handler requireAuth(Class<? extends Client> clientType, Authorizer
    * }</pre>
    *
    * @param ctx the handling context
-   * @param clientType the client type to authenticate with
+   * @param clientName the name of the client to authenticate with
    * @return a promise for the user profile, fulfilled if logged in
    */
-  public static Promise<UserProfile> login(Context ctx, Class<? extends Client> clientType) {
-    if (isDirect(clientType)) {
+  public static Promise<UserProfile> login(Context ctx, String clientName) {
```
Comment (gcoopercos) [inline src/main/java/ratpack/pac4j/RatpackPac4j.java:366]: I'm not familiar with the usages of this.  Over all, this PR looks like general upgrade work.  The biggest thing that stood out for me was the changing from using "clientType" to "clientName" and some of the 'isDirect(..)'  functional change.

But, I couldn't say what is "right". Seems ok.

File: pom.xml [hunk 2]
```diff
@@ -66,8 +66,8 @@
     </developers>
 
     <properties>
-        <pac4j.version>5.1.4</pac4j.version>
-        <java.version>11</java.version>
+        <pac4j.version>6.0.0</pac4j.version>
+        <java.version>17</java.version>
```
Comment (gcoopercos) [inline pom.xml:70]: Assuming the jdk switch is ok?

Comment (rnett) [reply pom.xml:70]: Yep, it was done in pac4j.

File: src/main/java/ratpack/pac4j/RatpackPac4j.java (hunk 1 above)
Comment (rnett) [reply src/main/java/ratpack/pac4j/RatpackPac4j.java:366]: Yeah, the client name is something that was forced by pac4j itself, there are no longer getters for the type.

**Review by jthurne — APPROVED** [review]

**Comment by rnett** [comment]
Closing to make the real PRs.

**Pull request closed without merging**